kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java

```diff
@@ -108,4 +121,69 @@ public void delete() {
   public void cleanup() {
     client.pods().inNamespace(currentNamespace).delete();
   }
+
+  @Test
+  public void log() throws InterruptedException {
+    waitUntilPodIsReady(pod2.getMetadata().getName(), 60);
+    String log = client.pods().inNamespace(currentNamespace).withName(pod2.getMetadata().getName()).getLog();
+    assertNotNull(log);
+  }
+
+  @Test
+  public void exec() throws InterruptedException {
+    waitUntilPodIsReady(pod1.getMetadata().getName(), 60);
+    final CountDownLatch execLatch = new CountDownLatch(1);
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ExecWatch execWatch = client.pods().inNamespace(currentNamespace).withName(pod1.getMetadata().getName())
+      .writingOutput(out).usingListener(new ExecListener() {
+        @Override
+        public void onOpen(Response response) {
+          logger.info("Shell was opened");
+        }
+
+        @Override
+        public void onFailure(Throwable throwable, Response response) {
+          logger.info("Shell barfed");
+          execLatch.countDown();
+        }
+
+        @Override
+        public void onClose(int i, String s) {
+          logger.info("Shell closed");
+          execLatch.countDown();
+        }
+      }).exec("date");
+
+    execLatch.await(10, TimeUnit.MINUTES);
+    assertNotNull(execWatch);
+    assertNotNull(out.toString());
+  }
+
+  /**
+   * A Simple utility function to watch over pod until it gets ready
+   *
+   * @param podName Name of the pod
+   * @param nAwaitTimeout Time in seconds upto which pod must be watched
+   * @throws InterruptedException
+   */
+  private void waitUntilPodIsReady(String podName, int nAwaitTimeout) throws InterruptedException {
+    final CountDownLatch readyLatch = new CountDownLatch(1);
+    try (Watch watch = client.pods().withName(podName).watch(new Watcher<Pod>() {
```

Comment:
It would be more readable if we can extract `new Watcher<Pod>()` into private class.